dependencies.yaml

```diff
@@ -0,0 +1,153 @@
+files:
+  all:
+    output: conda
+    matrix:
+      cuda: ["11.5"]
+      arch: [x86_64]
+    includes:
+      - build
+      - cudatoolkit
+      - checks
+      - doc
+      - py_version
+      - run
+      - test_python
+  test_python:
+    output: none
+    includes:
+      - cudatoolkit
+      - py_version
+      - test_python
+  checks:
+    output: none
+    includes:
+      - checks
+      - py_version
+channels:
+  - rapidsai
+  - rapidsai-nightly
+  - conda-forge
+  - nvidia
+dependencies:
+  build:
+    common:
+      - output_types: [conda, requirements]
+        packages:
+          - cmake>=3.23.1,!=3.25.0
+          - ninja
+      - output_types: conda
+        packages:
+          - c-compiler
+          - cxx-compiler
+    specific:
+      - output_types: conda
+        matrices:
+          - matrix:
+              arch: x86_64
+            packages:
+              - gcc_linux-64=9.*
+              - sysroot_linux-64==2.17
+          - matrix:
+              arch: aarch64
+            packages:
+              - gcc_linux-aarch64=9.*
+              - sysroot_linux-aarch64==2.17
+      - output_types: conda
+        matrices:
+          - matrix:
+              arch: x86_64
+              cuda: "11.5"
+            packages:
+              - nvcc_linux-64=11.5
+          - matrix:
+              arch: aarch64
+              cuda: "11.5"
+            packages:
+              - nvcc_linux-aarch64=11.5
+  checks:
+    common:
+      - output_types: [conda, requirements]
+        packages:
+          - pre-commit
+  cudatoolkit:
+    specific:
+      - output_types: conda
+        matrices:
+          - matrix:
+              cuda: "11.2"
+            packages:
+              - cudatoolkit=11.2
+          - matrix:
+              cuda: "11.4"
+            packages:
+              - cudatoolkit=11.4
+          - matrix:
+              cuda: "11.5"
+            packages:
+              - cudatoolkit=11.5
+  doc:
+    common:
+      - output_types: [conda, requirements]
+        packages:
+          - ipython
+          - nbsphinx
+          - numpydoc
+          - pydata-sphinx-theme
+          - recommonmark
+          - sphinx
+  py_version:
+    specific:
+      - output_types: conda
+        matrices:
+          - matrix:
+              py: "3.8"
+            packages:
+              - python=3.8
+          - matrix:
+              py: "3.9"
+            packages:
+              - python=3.9
+          - matrix:
+            packages:
+              - python>=3.8,<3.10
+  run:
+    common:
+      - output_types: conda
+        packages:
+          - click
+          - cupy >=10,<12.0.0a0
+          - jbig
+          - jpeg
+          - libwebp-base
+          - numpy
+          - scikit-image >=0.19.0,<0.20.0a0
+          - scipy
+          - xz
+          - zlib
+          - zstd
+          # Not sure where these go, if anywhere:
+          # - openslide
+          # - xorg-libxcb
+          # - yasm
```

Comment:
@jakirkham Related to your question about yasm above -- what section do these go in? Build deps? Runtime deps? Testing deps?

Reply:
YASM is an assembler. It is used for building

Am not aware of any runtime assembly occurring here. Though it is a fair question given CuPy does some runtime compilation, but it doesn't use YASM AFAIK

Reply:
Aha. I didn't see `yasm` in the libcucim conda recipe until now. I'll add it here as a requirement on x86_64, like in the conda recipe (though it's weird that only one platform would need it...).

What about `openslide` and `xorg-libxcb`? Are those runtime requirements? Test requirements?

Reply:
yasm updated in b91daf5e43a571567df7601e6446e06b37be800d.

Reply:
Sounds good. Thanks Bradley! 🙏

YASM is an `x86` & `x86_64` assembler. It wouldn't really make sense to use on ARM since it would generate instructions that don't run there (or do and cause havoc 😟)

I don't recall exactly where YASM is used by `libcucim`, but would not be surprised if there is a fallback path that works on any architecture (like some C code). YASM really comes in when one writes their own assembly for a function or set of functions. This can come in handy for some video encoding/decoding libraries, which are very performance sensitive.